### PR TITLE
Generate client and add tests for all personalizer v1.0 apis . Azure Specs PR - https://github.com/Azure/azure-rest-api-specs/pull/7624

### PR DESCRIPTION
**Pull request by dwaijam**
Generate client and add tests for all personalizer v1.0 apis

### CODE REVIEW

**Comment by shahabhijeet** [comment]
@dwaijam before this can be reviewed, can you add link to your REST api spec.
Also make sure you use generate.ps1 to generate your SDK, that ensures correct meta data generation, it is especially required for properly tagging your generated nuget package.

File: src/SDKs/CognitiveServices/dataPlane/Personalizer/Personalizer.Tests/Personalizer/ConfigurationsTests.cs [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        {
+            using (MockContext.Start(this.GetType().FullName))
+            {
+                HttpMockServer.Initialize(this.GetType().FullName, "ResetPolicy");
+
+                IPersonalizerClient client = GetClient(HttpMockServer.CreateInstance());
+
+                PolicyContract policy = await client.Policy.ResetAsync();
+
+                Assert.Equal("--cb_explore_adf --quadratic GT --quadratic MR --quadratic GR --quadratic ME --quadratic OT --quadratic OE --quadratic OR --quadratic MS --quadratic GX --ignore A --cb_type ips --epsilon 0.2",
```
Comment (shahabhijeet) [inline src/SDKs/CognitiveServices/dataPlane/Personalizer/Personalizer.Tests/Personalizer/ConfigurationsTests.cs]: nit: make this const

File: src/SDKs/CognitiveServices/dataPlane/Personalizer/Personalizer.Tests/Personalizer/ConfigurationsTests.cs [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+            using (MockContext.Start(this.GetType().FullName))
+            {
+                HttpMockServer.Initialize(this.GetType().FullName, "UpdatePolicy");
+
+                IPersonalizerClient client = GetClient(HttpMockServer.CreateInstance());
+
+                var policy = new PolicyContract
+                {
+                    Name = "app1",
+                    Arguments = "--cb_explore_adf --quadratic GT --quadratic MR --quadratic GR --quadratic ME --quadratic OT --quadratic OE --quadratic OR --quadratic MS --quadratic GX --ignore A --cb_type ips --epsilon 0.2"
```
Comment (shahabhijeet) [inline src/SDKs/CognitiveServices/dataPlane/Personalizer/Personalizer.Tests/Personalizer/ConfigurationsTests.cs]: nit: make this const

File: src/SDKs/CognitiveServices/dataPlane/Personalizer/Personalizer.Tests/Personalizer/EvaluationsTests.cs [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+{
+    public class EvaluationsTests : BaseTests
+    {
+        [Fact]
+        public void GetEvaluations()
+        {
+            using (MockContext.Start(this.GetType().FullName))
+            {
+                HttpMockServer.Initialize(this.GetType().FullName, "GetEvaluations");
+
```
Comment (shahabhijeet) [inline src/SDKs/CognitiveServices/dataPlane/Personalizer/Personalizer.Tests/Personalizer/EvaluationsTests.cs]: nit: any reason for adding blank lines? If there isn't a particular reason you should remove it.
When styleCop or static analyzers are enabled, you will be forced to fix this as all these warnings will be reported as errors.

File: src/SDKs/CognitiveServices/dataPlane/Personalizer/Personalizer.Tests/Properties/launchSettings.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+  "profiles": {
+    "Personalizer.Tests": {
+      "commandName": "Project",
+      "environmentVariables": {
+        "Key": "Value",
+        "AZURE_TEST_MODE": "Record"
```
Comment (shahabhijeet) [inline src/SDKs/CognitiveServices/dataPlane/Personalizer/Personalizer.Tests/Properties/launchSettings.json]: @dwaijam can you clarify why are you adding this file?
Are you setting testframework in record mode everytime you run these tests?
We do not want tests to run in record mode during CI
If you want more information, please follow testFramework documentation as these settings are only meant during development of tests and recording of tests locally.

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

**Comment by annelo-msft** [comment]
It looks like the file path for these files is incorrect ... they should live alongside the existing Personalizer files at https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/cognitiveservices/Personalizer

File: src/SDKs/CognitiveServices/dataPlane/Personalizer/Personalizer/Customizations/PersonalizerClientExtensions.cs [hunk 5]
```diff
@@ -0,0 +1,89 @@
+using Microsoft.Azure.CognitiveServices.Personalizer.Models;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
```
Comment (AlexGhiondea) [inline src/SDKs/CognitiveServices/dataPlane/Personalizer/Personalizer/Customizations/PersonalizerClientExtensions.cs]: I think we should add a copyright header to these files.

**Review by AlexGhiondea — APPROVED** [review]

File: src/SDKs/CognitiveServices/dataPlane/Personalizer/Personalizer.Tests/Personalizer/ConfigurationsTests.cs (hunk 1 above)
Comment (dwaijam) [reply src/SDKs/CognitiveServices/dataPlane/Personalizer/Personalizer.Tests/Personalizer/ConfigurationsTests.cs]: wont fix

File: src/SDKs/CognitiveServices/dataPlane/Personalizer/Personalizer.Tests/Personalizer/ConfigurationsTests.cs (hunk 2 above)
Comment (dwaijam) [reply src/SDKs/CognitiveServices/dataPlane/Personalizer/Personalizer.Tests/Personalizer/ConfigurationsTests.cs]: wont fix

File: src/SDKs/CognitiveServices/dataPlane/Personalizer/Personalizer.Tests/Personalizer/EvaluationsTests.cs (hunk 3 above)
Comment (dwaijam) [reply src/SDKs/CognitiveServices/dataPlane/Personalizer/Personalizer.Tests/Personalizer/EvaluationsTests.cs]: fixed

File: src/SDKs/CognitiveServices/dataPlane/Personalizer/Personalizer.Tests/Properties/launchSettings.json (hunk 4 above)
Comment (dwaijam) [reply src/SDKs/CognitiveServices/dataPlane/Personalizer/Personalizer.Tests/Properties/launchSettings.json]: removed

File: src/SDKs/CognitiveServices/dataPlane/Personalizer/Personalizer/Customizations/PersonalizerClientExtensions.cs (hunk 5 above)
Comment (dwaijam) [reply src/SDKs/CognitiveServices/dataPlane/Personalizer/Personalizer/Customizations/PersonalizerClientExtensions.cs]: fixed

**Review by shahabhijeet — APPROVED** [review]

**Review by AlexGhiondea — CHANGES_REQUESTED** [review]
There was some big changes between the time i signed off and the latest version. I have left a couple of questions.

**Comment by dwaijam** [comment]
> There was some big changes between the time i signed off and the latest version. I have left a couple of questions.

Addressed your concerns. This is a new version of the API. We are moving from preview to v1.0 and these changes are coming from generating the code from a new swagger which has been approved by the Azure API review board

**Review by AlexGhiondea — APPROVED** [review]

**Pull request merged**